packages/cli/src/ui/components/SettingsDialog.test.tsx

```diff
@@ -152,8 +151,32 @@ vi.mock('../../utils/settingsUtils.js', async () => {
 // const originalConsoleError = console.error;
 
 describe('SettingsDialog', () => {
+  // Simple delay function for remaining tests that need gradual migration
   const wait = (ms = 50) => new Promise((resolve) => setTimeout(resolve, ms));
 
+  // Custom waitFor utility for ink testing environment (not compatible with @testing-library/react)
```

Comment:
for posterity, i looked into this a bit more and (correct me if I'm wrong @bulkypanda !) but `@testing-library/react` was designed to hook into the DOM to know when elements are ready, which is not relevant or useful here since we're working with a command line UI rendered via Ink